Classes/EventListener/ImportSingleIcalEventListener.php

```diff
@@ -54,12 +54,13 @@ public function __invoke(ImportSingleIcalEvent $event): void
      * Initializes or gets an event by import id.
      *
      * @param string $importId
+     * @param int $pid
      *
      * @return Event
      */
-    private function initializeEventRecord(string $importId): Event
+    private function initializeEventRecord(string $importId,$pid): Event
     {
-        $eventObj = $this->eventRepository->findOneByImportId($importId);
+        $eventObj = $this->eventRepository->findOneByImportId($importId,$pid);
```

Comment:
The function `findOneByImportId` does not have an argument PID. 
You forgot to add your changes for the EventRepository.

Reply:
Thanx, I must have missed that my local patch had failed hunks as it was against 12.4.4 instead of master.

Reply:
EventRepository changes are added now.